PurchasesCoreSwift/FoundationExtensions/Dictionary+Extensions.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  Dictionary+Extensions.swift
+//  PurchasesCoreSwift
+//
+//  Created by César de la Vega on 7/21/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+extension Dictionary {
+
+    func removingNSNullValues() -> Dictionary {
+        self.filter { !($0.value is NSNull) }
+    }
+
+}
+
+extension Dictionary {
+
+    /// Merge strategy to use for any duplicate keys.
+    enum MergeStrategy<Value> {
+        /// Keep the original value.
+        case keepOriginalValue
+        /// Overwrite the original value.
+        case overwriteValue
+
+        var combine: (Value, Value) -> Value {
+            switch self {
+            case .keepOriginalValue:
+                return { original, _ in original }
+            case .overwriteValue:
+                return { _, overwrite in overwrite }
+            }
+        }
+    }
+
+    /// Creates a dictionary by merging the given dictionary into this
+    /// dictionary, using a merge strategy to determine the value for
+    /// duplicate keys.
+    ///
+    /// - Parameters:
+    ///   - other:  A dictionary to merge.
+    ///   - strategy: The merge strategy to use for any duplicate keys. The strategy provides a
+    ///   closure that returns the desired value for the final dictionary. The default is `overwriteValue`.
+    /// - Returns: A new dictionary with the combined keys and values of this
+    ///   dictionary and `other`.
+    func merging(_ other: [Key: Value], strategy: MergeStrategy<Value> = .overwriteValue) -> [Key: Value] {
+        merging(other, uniquingKeysWith: strategy.combine)
+    }
+
+    /// Merge the keys/values of two dictionaries.
+    ///
+    /// The merge strategy used is `overwriteValue`.
+    ///
+    /// - Parameters:
+    ///   - lhs: A dictionary to merge.
+    ///   - rhs: Another dictionary to merge.
+    /// - Returns: An dictionary with keys and values from both.
+    static func + (lhs: [Key: Value], rhs: [Key: Value]) -> [Key: Value] {
```

Comment:
I've added this method. I think it's not needed but sometimes it could be useful to reduce the amount of code for large methods. WDYT?

Reply:
I personally love it! I've wanted to use + many times myself. I'm the least experienced with Swift on this team though, so I'll let @aboedo or @taquitos weigh in :)

Reply:
freaking love it ❤️